src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceFacade.java

```diff
@@ -164,6 +166,38 @@ public void putDatasource(final Datasource datasource, final ActionListener list
         });
     }
 
+    /**
+     * Delete datasource in an index {@code DatasourceExtension.JOB_INDEX_NAME}
+     *
+     * @param datasource the datasource
+     *
+     */
+    public void deleteDatasource(final Datasource datasource) {
+        if (client.admin()
+            .indices()
+            .prepareDelete(datasource.getIndices().toArray(new String[0]))
+            .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN)
+            .execute()
+            .actionGet(clusterSettings.get(Ip2GeoSettings.TIMEOUT))
+            .isAcknowledged() == false) {
+            throw new OpenSearchException("failed to delete data[{}] in datasource", String.join(",", datasource.getIndices()));
+        }
+        DeleteResponse response = client.prepareDelete()
+            .setIndex(DatasourceExtension.JOB_INDEX_NAME)
+            .setId(datasource.getName())
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .execute()
+            .actionGet(clusterSettings.get(Ip2GeoSettings.TIMEOUT));
+
+        if (response.status().equals(RestStatus.OK)) {
+            log.info("deleted datasource[{}] successfully", datasource.getName());
+        } else if (response.status().equals(RestStatus.NOT_FOUND)) {
+            throw new ResourceNotFoundException("datasource[{}] does not exist", datasource.getName());
+        } else {
+            throw new OpenSearchException("failed to delete datasource[{}] with status[{}]", datasource.getName(), response.status());
+        }
```

Comment:
I would like to keep the code as it. Having them in a single if/ifelse/else statement makes sense to me because they are all checking value on the same variable. reponse.status().